### PR TITLE
Implement hasReply avoid copying the shared_ptr just to peak at the r

### PR DESCRIPTION
**Pull request by baldersheim**
…esult.

@vekterli PR

### CODE REVIEW

File: storage/src/vespa/storage/persistence/persistenceutil.h [hunk 1]
```diff
@@ -43,20 +42,27 @@ class MessageTracker : protected Types {
...
-    api::StorageReply::SP getReply() {
-        return _reply;
+    bool hasReply() const { return bool(_reply); }
+    const api::StorageReply & getReply() const {
+        return *_reply;
+    }
+    api::StorageReply & getReply() {
+        return *_reply;
+    }
+    api::StorageReply::SP getReplySP() {
```
Comment (vekterli) [inline storage/src/vespa/storage/persistence/persistenceutil.h:52]: I'd prefer this to be called something like `steal_reply_sp`, as the current name implies the function doesn't have any side effects

**Review by vekterli — APPROVED** [review]
:+1: one comment for consideration

File: storage/src/vespa/storage/persistence/persistenceutil.h (hunk 1 above)
Comment (toregge) [inline storage/src/vespa/storage/persistence/persistenceutil.h:52]: Please use an rvalue qualifier for this method.

Comment (baldersheim) [reply storage/src/vespa/storage/persistence/persistenceutil.h]: Done.

**Review by baldersheim — COMMENTED** [review]
?

**Pull request merged**